pkg/manifest/os.go

```diff
@@ -1162,3 +1175,64 @@ func (p *OS) addInlineDataAndStages(pipeline *osbuild.Pipeline, files []*fsnode.
 		p.inlineData = append(p.inlineData, string(file.Data()))
 	}
 }
+
+// versionLockShim generates an osbuild stage and inline file to create a dnf
+// versionlock configuration file that locks the shim package to its installed
+// version. This is necessary for UKI-based images that can become unbootable
+// if the shim is upgraded to one signed by a newer certificate that does not
+// exist in the system's database.
+// See https://issues.redhat.com/browse/RHEL-93650
+func versionLockShim(packages []rpmmd.PackageSpec, a arch.Arch) (*fsnode.File, error) {
```

Comment:
Nitpick: if this becomes the way to lock version for packages, it would probably make sense to extract the creation of the `*fsnode.File`  for a given array of `PackageSpec`s, to the `versionlock` package. And keep only the `shim-x64` part here.